includes/discovery/sensors/temperature/comware.inc.php

```diff
@@ -55,16 +60,16 @@
 foreach ($pre_cache['comware_oids'] as $index => $entry) {
     if (is_numeric($entry['hh3cTransceiverTemperature']) && $entry['hh3cTransceiverTemperature'] != 2147483647 && isset($entry['hh3cTransceiverDiagnostic'])) {
         $oid = '.1.3.6.1.4.1.25506.2.70.1.1.1.15.' . $index;
-        $dbquery = dbFetchRows("SELECT `ifDescr` FROM `ports` WHERE `ifIndex`= ? AND `device_id` = ? AND `ifAdminStatus` = 'up'", [$index, $device['device_id']]);
         $limit_low = $entry['hh3cTransceiverTempLoAlarm'] / $divisor_alarm;
         $warn_limit_low = $entry['hh3cTransceiverTempLoWarn'] / $divisor_alarm;
         $limit = $entry['hh3cTransceiverTempHiAlarm'] / $divisor_alarm;
         $warn_limit = $entry['hh3cTransceiverTempHiWarn'] / $divisor_alarm;
         $current = $entry['hh3cTransceiverTemperature'];
         $entPhysicalIndex = $index;
         $entPhysicalIndex_measured = 'ports';
-        foreach ($dbquery as $dbindex => $dbresult) {
```

Comment:
was this foreach useless?

Reply:
hi @Jellyfrog 
that was one big question, but test data is passing
from my point of view, ifIndex is unique for each port and could not be two port indexed with same value
so this query (original)
$dbquery = dbFetchRows("SELECT `ifDescr` FROM `ports` WHERE `ifIndex`= ? AND `device_id` = ? AND `ifAdminStatus` = 'up'", [$index, $device['device_id']
simply return array in array, that was the reason for "foreach" to extract array